### ISSUE TITLE
UDv2 documentation of questions and relative clauses

### ISSUE THREAD

**Issue by nschneid** [UniversalDependencies/docs#454]
As far as I can tell, there is not really any discussion of questions (yes/no or wh-) in the UDv2 guidelines. Should a section on questions be added? I have received inquiries about, e.g., what should be the head of the sentence in "How are you doing?".

**Comment by jnivre** [comment]
A section of questions seems like a good idea. However, the head should be the main verb ("doing") as usual:

nsubj(doing, you)
aux(doing, are)
advmod(doing, how)

**Comment by nschneid** [comment]
Towards the goal of drafting documentation of questions, wh-words, and relative clauses, some examples:

1. Do you like my hat?
2. I wonder whether/if you like it.
3. I wonder why you don't like it.
4. Why don't you like it?
5. What don't you like about it?
6. What I don't like is the color.
7. What I don't like about it is the color.
8. I don't like what color it is.
9. What do you care about? (preposition stranding)
10. Is that what you care about? (preposition stranding)
11. I don't care what you like.
12. I don't care what color you like.
13. I don't care what color you wish it was. 
14. a hat which/that/∅ you will like
15. a hat which/that will satisfy you
16. a hat which/that/∅ I am sure (that) you will like
17. a hat which/that/∅ I am sure (*that) will satisfy you
18. a friend who/that/∅ I am sure (that) you will like (if different from 16)
19. skills which/that/∅ I need to get a job

Presumably the [v1 English specific syntax](http://universaldependencies.org/en/overview/specific-syntax.html#clausal-modifiers-of-nouns) page holds some of the answers.

**Comment by nschneid** [comment]
1: just like "You do like my hat"

2: "whether"/"if" is always `mark`

3: Not sure how to decide if "why" heads "like" or vice versa.

4: `ccomp(matrix-verb, embedded-verb)`, and "why" attaches to the embedded verb

5, 7: Is "what about it" a constituent? Or is "about it" an independent oblique of "like"?

6: pseudo-cleft: `nsubj(color, What)`, `acl:relcl(What, like)`? The [v1 English specific syntax page](http://universaldependencies.org/en/overview/specific-syntax.html#basic-analysis) gives "What the committee hopes to learn is why..." as `nsubj(why, What)`. However, the [v2 specific syntax page](http://universaldependencies.org/u/overview/complex-syntax.html#clausal-subjects) gives "What she said is interesting" as `csubj(interesting, said)`. Is this a contradiction? Or would "What she said is that she likes my hat" be different? 

8: `det(color, what)`, `nsubj(color, it)`?

9: `obl(care, what)`, `case(what, about)`

10: `acl:relcl(what, care)`, `obl(care, about)`, `nsubj(what, that)`? (Related Yodaism: "That is why you fail.")

11-13: should follow from above

14-19: `acl:relcl(noun, embedded-predicate)`. Normally, relativizer (if overt) is a subject, object, or oblique of the embedded predicate. Normally, "I am sure that X" seems to be treated as `nsubj(sure, I)` and `ccomp(sure, X)`. So does "sure" (as the embedded predicate) end up with multiple core dependents here, or does the relativizer hop over the "sure" and attach to its complement's verb? Also, in 16, is it still `ccomp` if "that" is forbidden?

**Comment by nschneid** [comment]
Related: #326, UniversalDependencies/UD_English#23

**Comment by nschneid** [comment]
Probably worth updating as part of this: neither [`acl`](http://universaldependencies.org/docs/u/dep/acl.html) nor [complex clauses](http://universaldependencies.org/u/overview/complex-syntax.html) mentions `acl:relcl`. [Nominal syntax](http://universaldependencies.org/u/overview/nominal-syntax.html) does, and I've added an example to illustrate a relative pronoun, but there is no real discussion.

**Comment by nschneid** [comment]
I've updated [`acl`](http://universaldependencies.org/docs/u/dep/acl.html), adding an explicit mention of `acl:relcl` and several more examples from [v1 English syntax](http://universaldependencies.org/en/overview/specific-syntax.html#basic-analysis).

**Comment by dan-zeman** [comment]
We lack conventions for mentioning language-specific relations in the documentation. `acl:relcl` is not supposed to be documented in the universal guidelines because it doesn't have the universal status.

On the other hand, it is obvious that this and many other language-specific subtypes are usable in a large number of languages and should be documented somewhere centrally so that people can apply them to new languages.

**Comment by martinpopel** [comment]
I think sooner or later we will need to abandon the idea that deprel subtype = language specific subtype.
Is anyone aware of a language where `acl:relcl` is not applicable? [Wikipedia](https://en.wikipedia.org/wiki/Relative_clause#Examples) mentions almost all UD languages including those missing in [those already annotated](http://universaldependencies.org/ext-dep-index.html#acl). I think promoting `acl:relcl` to the main documentation is a good idea, which can improve consistency across languages.
I think we don't need to explicitly distinguish which deprel subtypes are language specific and which are (almost) universal.

Note that even the main deprels are not "universal" in a sense that they are present in all treebanks. Some (e.g. `clf`) are not even applicable to all languages. UD2.0 contains 70 treebanks, but some deprels are used only in few treebanks:
```
1  clf
8  reparandum
9  list
16  goeswith
22  dislocated
37  expl
39  orphan
40  discourse
43  vocative
46  dep
50  compound
52  fixed
57  flat
59  parataxis
```
Note that `acl:relcl` is present in 32 treebanks.

**Comment by amir-zeldes** [comment]
I agree that relative clauses are probably present in just about any language in some form. The examples above aren't all missing from treebanks for the same reasons though:

  * `reparandum` and `goeswith` depend on whether dysfluencies/orthographic errors appear in the data, so this doesn't depend on the language so much as the underlying text type. Similarly `list` has to do with data from sources that have overt lists.
  * I suspect that `flat`, `fixed`, `discourse` and `parataxis` correspond to universal phenomena, but individual guidelines differ on whether/how to use them
  * I think `vocative` is not explicitly marked in many languages, but is functionally 100% a universal category. I'm not as sure about this, but I also expect `orphan` is universal, but rare enough that it doesn't appear in many TBs (or is ignored since it's so rare)

As potentially language specific I would mostly expect: 

  * clf
  * compound
  * dislocated (though probably possible in the majority of languages)
  * expl

And `dep` is just a matter of how well the guidelines cover the data (and accordingly maybe a function of data size and diversity, or also whether or not an automatic conversion produces it).

**Comment by martinpopel** [comment]
Yes, I agree with @amir-zeldes' analysis. I guess there are many treebanks where a deprel xy is missing rather for legacy reasons (semi-automatic conversion from older annotation, not enough time to re-annotate the whole data). But this was not my point (the data will never be perfect and I am OK with this).
I am not suggesting to make clf/compound/... language-specific. I am suggesting to document and standardize even the deprel subtypes (and try to use them in as many treebanks as possible).

**Comment by amir-zeldes** [comment]
Yes, agreed, in the case of `acl:relcl` it is definitely more of a universal relation. Maybe we should suggest a list of the subtypes that deserve general (non language specific) documentation. From the English subtypes I'd also nominate:

  * nmod:poss
  * cc:preconj

Both of these appear in many languages and especially the way that cc:preconj works should intuitively be standardized across languages (poss is maybe more trivial)

**Comment by jnivre** [comment]
This is one of many things that we need to address when reworking the documentation.

**Comment by dan-zeman** [comment]
Agreed that we need documentation for common types of relation subtypes.

I, too, think that `acl:relcl` is more universal than `clf`. What makes `clf` deserve its own top-level label is that it is not easy to come up with another universal relation (other than `dep`) of which it could be a subtype.

**Comment by dan-zeman** [comment]
A draft of [documentation of relative clauses is being developed here](https://universaldependencies.org/workgroups/newdoc/relative_clauses.html).
